website/pages/en/developing/assemblyscript-api.mdx

```diff
@@ -280,6 +280,20 @@ As the entity may not exist in the store yet, the `load` method returns a value
 
 > **Note:** Loading entities is only necessary if the changes made in the mapping depend on the previous data of an entity. See the next section for the two ways of updating existing entities.
 
+#### Looking up entities created withing a block
+
+The store API facilitates the retrieval of entities that were created or updated in the current block. This feature proves to be useful in situations where the handler requires the on-chain event to be generated in the same block. By utilizing this functionality, indexing performance can improve significantly.
```

Comment:
> This feature proves to be useful in situations where the handler requires the on-chain event to be generated in the same block

I think here we can borrow from @lutter's description from the graph-node PR...

A typical situation for this is that one handler creates a Transaction from some on-chain event, and a later handler wants to access this transaction if it exists. In the case where the transaction does not exist, the subgraph will have to go to the database just to find out that the entity does not exist; if the subgraph author already knows that the entity must have been created in the same block, using `loadInBlock` avoids this database roundtrip. For some subgraphs, these missed lookups can contribute significantly to the indexing time.

Reply:
I think also worth noting that if there is no entity created "in block" (even if there **is** an entity with the right ID in the database), this will return `null`